packages/frontend/src/components/DonateToUkraine/DonateContainerV2.js

```diff
@@ -0,0 +1,207 @@
+import BN from 'bn.js';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import { Mixpanel } from '../../mixpanel/index';
+import FungibleTokens from '../../services/FungibleTokens';
+import classNames from '../../utils/classNames';
+import isDecimalString from '../../utils/isDecimalString';
+import { getNearAndFiatValue } from '../common/balance/helpers';
+import Container from '../common/styled/Container.css';
+import EnterAmount from '../send/components/views/EnterAmount';
+import SelectToken from '../send/components/views/SelectToken';
+import Success from '../send/components/views//Success';
+
+const { getFormattedTokenAmount, getParsedTokenAmount, getUniqueTokenIdentity } = FungibleTokens;
+
+export const VIEWS = {
+    ENTER_AMOUNT: 'enterAmount',
+    SELECT_TOKEN: 'selectToken',
+    SUCCESS: 'success'
+};
+
+const StyledContainer = styled(Container)`
+    &&& {
+        .header {
+            position: relative;
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            color: #272729;
+            font-weight: 600;
+            font-size: 20px;
+            word-break: break-all;
+
+            .back-arrow-button {
+                position: absolute;
+                left: 0;
+            }
+        }
+
+        .buttons-bottom-buttons {
+            margin-top: 55px;
+
+            > button {
+                display: block;
+                width: 100%;
+            }
+
+            .link {
+                display: block;
+                margin: 20px auto;
+            }
+        }
+
+        @media (max-width: 500px) {
+            .buttons-bottom {
+                display: flex;
+                flex-direction: column;
+                min-height: calc(100vh - 160px);
+
+                .buttons-bottom-buttons {
+                    margin-top: auto;
+                }
+            }
+
+            &.showing-banner {
+                .buttons-bottom {
+                    min-height: calc(100vh - 218px);
+                }
+            }
+        }
+    }
+`;
+
+const DonateContainerV2 = ({
+    redirectTo,
+    fungibleTokens,
+    accountId,
+    isMobile,
+    explorerUrl,
+    showNetworkBanner,
+    activeView,
+    setActiveView,
+    handleSendToken,
+    sendingToken,
+    transactionHash,
+    nearTokenFiatValueUSD
+}) => {
+    const [userInputAmount, setUserInputAmount] = useState('');
+    const [isMaxAmount, setIsMaxAmount] = useState(false);
+
+    const [selectedToken, setSelectedToken] = useState(fungibleTokens[0]);
+
+    useEffect(() => {
+        let targetToken = fungibleTokens.find(({ contractName }) => 
```

Comment:
nit: this is hard to read imo might be worth it to break out into a common function as both sides of this Maybe are 80% the same.